datumaro/components/hl_ops.py

```diff
@@ -99,8 +100,13 @@ def merge(*datasets: IDataset) -> IDataset:
     """
 
     categories = ExactMerge.merge_categories(d.categories() for d in datasets)
+    media_type = datasets[0].media_type()
+    for d in datasets:
+        if d.media_type() is not media_type:
+            raise MediaTypeError("Datasets have different media types")
```

Comment:
It would be better to put this into `ExactMerge.merge`

Reply:
Moved this to `ExactMerge.merge_media_types`

Reply:
Why not into `merge`? It is the essential check for this operation.

Reply:
Because need to give `media_type` in `DatasetItemStorageDatasetView`. Could this be a problem?